### PR TITLE
Remove broken link

### PR DESCRIPTION
**Pull request by kmaid**
Hopefully pass through undefined links

### CODE REVIEW

File: apps/crn-server/src/controllers/research-output.controller.ts [hunk 1]
```diff
@@ -144,6 +146,20 @@ export default class ResearchOutputController {
...
+        link: currentResearchOutput?.link,
+        type: currentResearchOutput?.type,
+        documentType:
+          currentResearchOutput?.documentType &&
+          isResearchOutputDocumentType(currentResearchOutput?.documentType)
+            ? currentResearchOutput?.documentType
+            : 'Grant Document',
+      };
+    }
+
```
Comment (Riccardo-Brown) [inline apps/crn-server/src/controllers/research-output.controller.ts:162]: i think the flow of the code would be better if this code was below the  `currentResearchOutput` check. 

Also should we be checking the RO is published before creating the version

File: apps/crn-server/src/controllers/research-output.controller.ts [hunk 2]
```diff
@@ -144,6 +146,20 @@ export default class ResearchOutputController {
     const currentResearchOutput =
       await this.researchOutputDataProvider.fetchById(id);
 
+    let version: ResearchOutputVersionPostRequest | undefined;
+    if (researchOutputUpdateData.createVersion) {
+      version = {
+        title: currentResearchOutput?.title || '',
```
Comment (gabiayako) [inline apps/crn-server/src/controllers/research-output.controller.ts:152]: shouldn't we send the new data here?

Comment (kmaid) [reply apps/crn-server/src/controllers/research-output.controller.ts:152]: No, when we make a new version that will replace the existing RO and the existing RO is transformed into the version.

Comment (oratiu) [reply apps/crn-server/src/controllers/research-output.controller.ts:152]: I knew I'd find the answer somehow

**Pull request merged**